dashboard/app/helpers/script_levels_helper.rb

```diff
@@ -8,11 +8,15 @@ def script_level_solved_response(response, script_level)
 
         # End-of-Stage Experience is only enabled for:
         # stages except for the last stage of a script
-        # users in sections with an enabled "stage extras" flag
+        # users in or teaching sections with an enabled "stage extras" flag
         enabled_for_stage = !script_level.end_of_script?
         enabled_for_user = current_user && current_user.section_for_script(script_level.script) &&
             current_user.section_for_script(script_level.script).stage_extras
-        if enabled_for_stage && enabled_for_user
+        enabled_for_teacher = current_user.try(:teacher?) &&
+            current_user.sections.any? do |section|
```

Comment:
nit: I think you could do `current_user.sections.where(script_id: script_level.script_id).any?` to make this a bit cheaper

Reply:
Turns out `stage_extras` is a column too, not a serialized property, so the query can take care of both.